distributed-lock-state-machine/distributed-lock-state-machine.md

```diff
@@ -1,15 +1,22 @@
 # Implementing a Distributed Lock State Machine
-Recently, I've been working with applications that construct Google Sheets with user-driven data.  The spreadsheet link is given to the user and the users can update the contents of the worksheet followed by clicking a hyperlink to effectively "send back" the data to the server for processing.
+A complex distributed computing problem application developers may encounter is controlled access to a resource or entity in a multi-user, concurrency environment.  How can we assuredly guard against an arbitrary number of concurrent processes from potentially mutating or even having any access to an entity simulatenously?  One possible solution is to rely on database management systems which provide concurrency safeguards via ACID transactions and row-level locking.
 
 In this article, you'll learn how I combined various technologies like Postgres, SQL and Node.js and created an application which implements a Distributed Lock State Machine.
 
+In understanding the approaches taken here, you'll learn how commitment control, isolation levels, and ACID transaction guarantees can be applied to implement a Distributed Lock State Machine.  In this context, "distributed" means multiple concurrent processes.  Moreover, "lock" is the safety mechanism that safeguards against concurrent processes from gaining access to shared resources or entities simultaneously.  Finally, "state machine" means those concurrent processes will have the ability to identify a resource or entity in a specific state (i.e. `new`, `in-progress`), perform some work or processing on them and transition them to a final, settled state (i.e `complete`, `error`).
```

Comment:
Minor suggestions:

- Maybe lower-case `Distributed Lock State Machine`?
- Remove `Moreover, `
- Oxford comma at `them and` -> `them, and`
- Should be `e.g.` instead of `i.e.`. My favorite formal grammar resource on the subject is https://www.youtube.com/watch?v=D_PL8ouEAus (pardon the profanity)